### DISCUSSION TITLE
How to change the foreground/background colors of LaTeX equations exported to svg using dvisvgm?

### DISCUSSION THREAD

**Discussion by kaushalmodi** [kaushalmodi/ox-hugo#612] [Q&A] [answered]
This discussion is created from https://github.com/kaushalmodi/ox-hugo/issues/611 by @Perangelot.
Below text mostly a copy of that original issue (only the details related to the SVG color are copied).

---

## Actual Behavior
Using the option `#+tex:dvisvgm`, LaTeX images are created, which works fine in general. 

The svg images are, unlike the usual `org-preview-images`, not generated with the emacs theme colors, but in black with no background. This is, in principle, what you want, but those who use a dark theme like me need other colors (I, for example, need the font color to be light gray). 

## Expected Behavior

Have customizable variables governing the foreground and background colors of the images. 

## How to Reproduce the Issue
Try this MWE:
```
#+title: tryout svg
#+startup: latexpreview
#+options: tex:dvisvgm

Let us try an inline formula like $p \to q$. Does this work? $r \to s$ 

Does this =tikz= picture work?

$\begin{tikzpicture}
  \draw (9,3) -- (1,4);
\end{tikzpicture}$
```
In org-mode:
![grafik](https://user-images.githubusercontent.com/89665177/161958591-85aa4108-f74a-42d1-a9c1-e2499acd8cf1.png)
HTML output:
![grafik](https://user-images.githubusercontent.com/89665177/161958912-65773f0c-07fd-43df-a567-e25a18e0f7e0.png)

**Comment by kaushalmodi** [comment]
@Perangelot

> The svg images are, unlike the usual org-preview-images, not generated with the emacs theme colors, but in black with no background. 

I am afraid this is out of my expertise. Also, this wouldn't be something specific to ox-hugo. You should be getting the same colors whether you exported using ox-html or any other backend. If you can figure out how to export with custom colors using ox-html, the same would work for ox-hugo as well. 

Trying asking this question on customizing dvisvgm image background and foreground colors on the Org mode mailing list. 

If you find out how to do that, let me know in comments so that it gets documented for other users.

**Comment by kaushalmodi** [comment, marked as answer]
@Perangelot

I looked around in the Org manual and I found this node [(org) Previewing LaTeX fragments](https://orgmode.org/manual/Previewing-LaTeX-fragments.html) -- There a variable `org-format-latex-options` is documented.

Upon doing `C-h v org-format-latex-options`, you will see:

```
org-format-latex-options is a variable defined in ‘org.el’.

Its value is
(:foreground default :background default :scale 1.0 :html-foreground "Black" :html-background "Transparent" :html-scale 1.0 :matchers
             ("begin" "$1" "$" "$$" "\\(" "\\["))

Options for creating images from LaTeX fragments.
This is a property list with the following properties:
:foreground  the foreground color for images embedded in Emacs, e.g. "Black".
             ‘default’ means use the foreground of the default face.
             ‘auto’ means use the foreground from the text face.
:background  the background color, or "Transparent".
             ‘default’ means use the background of the default face.
             ‘auto’ means use the background from the text face.
:scale       a scaling factor for the size of the images, to get more pixels
:html-foreground, :html-background, :html-scale
             the same numbers for HTML export.
:matchers    a list indicating which matchers should be used to
             find LaTeX fragments.  Valid members of this list are:
             "begin" find environments
             "$1"    find single characters surrounded by $.$
             "$"     find math expressions surrounded by $...$
             "$$"    find math expressions surrounded by $$....$$
             "\("    find math expressions surrounded by \(...\)
             "\["    find math expressions surrounded by \[...\]

  You can customize this variable.
```

---

The relevant pieces are:

> :html-foreground "Black" :html-background "Transparent" :html-scale 1.0

> :html-foreground, :html-background, :html-scale
             the same numbers for HTML export.

- `org-format-latex-options` is a plist.
- You can customize the foreground and background colors in the generated SVG/PNG when exporting to an HTML derived backend (`ox-hugo` extends indirectly from `ox-html`!) by modifying the `:html-foreground` and `:html-background` properties.

### Default LaTeX -> dvisvgm -> SVG foreground/background

From https://ox-hugo.scripter.co/test/posts/equations-exported-to-svg/, the equations show up in black foreground with transparent background by default:

![image](https://user-images.githubusercontent.com/3578197/162004601-e21fba6b-c425-468a-9327-2a133ada7c9d.png)

### SVG foreground/background after customization

If you evaluate this:

```elisp
(with-eval-after-load 'org
  (plist-put org-format-latex-options :html-foreground "Yellow")
  (plist-put org-format-latex-options :html-background "Blue"))
```

you will get:

![image](https://user-images.githubusercontent.com/3578197/162005251-8c15bccc-d5d0-42e4-bab7-ea27c056df0d.png)

**_Yes, it's not pretty -- I am just demonstrating the customization of SVG foreground/background_**.

_PS: The fonts are different because this second screenshot is from my localhost server where the default font type happens to be Serif._

**Reply by Perangelot** [reply]
Wow, thanks for the efforts! I had actually seen that variable myself but must have overlooked the attributes in that list. Since the formulae were placed too high, I added 
``` margin-bottom: -3.6px; !important;```
to the respective css class. Now it it perfectly aligned with the usual text:

![grafik](https://user-images.githubusercontent.com/89665177/162175207-1394f875-34a2-4f41-9aaa-c53de34ee44e.png)

This looks awesome now! I cannot stress how much you've helped me out. Thank you!
